dungeon-crawler/game_engine/src/main/java/com/novoda/dungeoncrawler/Conveyor.java

```diff
@@ -3,40 +3,42 @@
 import static com.novoda.dungeoncrawler.Direction.RIGHT_TO_LEFT;
 
 class Conveyor {
-    private boolean alive;
```

Comment:
`alive` had dual meanings in the codebase

1: as we expected if the enemy was dead or alive

2: things that where dead could be overriden in the array pool of enemies, we now avoid that and allocate all the space we need upfront